### DISCUSSION TITLE
Multiple detail page

### DISCUSSION THREAD

**Discussion by archana-multi-concept** [pimcore/pimcore#10024] [Q&A / Support]
Hello,

Facing issue with multiple detail page. In application there are two module one is blog and other one is news. I want to show detail pages of both blog and news. I followed pimcore demo coding pattern. From which I have created to link generator one for news and other one is for blog. For blog every thing is working fine, but when I click on news detail link. It redirect to blog detail page.
Not understand issue. I cross checked code multiple time. there is no replication in code.
For more information, please find attachment.
![Screenshot from 2021-08-13 12-31-22](https://user-images.githubusercontent.com/82560317/129317651-4de15711-292e-4e7d-be76-ddb9b111075e.png)

**Comment by archana-multi-concept** [comment]
issue is in route.

for news detail page route is:-
 @Route("{path}/{newstitle}~n{news}", name="news-detail", defaults={"path"=""}, requirements={"path"=".*?", "newstitle"="[\w-]+", "news"="\d+"})

and for blog-detail it is:- 
@Route("{path}/{blogtitle}~n{blog}", name="blog-detail", defaults={"path"=""}, requirements={"path"=".*?", "blogtitle"="[\w-]+", "blog"="\d+"})

How can I differentiate route for both detail page.

**Reply by passioneight** [reply]
Instead of `~n` use `~b` for your _blog_ route.

**Reply by archana-multi-concept** [reply]
Thanks for reply, I just resolved the issue with same solution.
Thanks again.
